### ISSUE TITLE
Add version information to the ServerAd structure

### ISSUE THREAD

**Issue by turetske** [PelicanPlatform/pelican#1725]
Pelican Service:
- [X] Origin
- [X] Cache

As part of getting origin versions from the prometheus metrics at the director, the director needs to have some idea of origin versions.

Generally, the way the Director gets its information from the Origin is via the Origin advertising information to the director. So additional information needs to be added to the ServerAd structure. This will also allow us to advertise version information from Cache's as well, since this would be general server information

Short version: Add a field in the ServerAd struct (in `server_structs/director.go`) that reports a version in the form of `v<x>.<y>.<z>` (eg. v7.11.10) and update the Origin and Cache code to populate this information on startup. Look in `launchers/` to see where the startup code is.
